siloLapisTests/test/protectedRoutes.spec.ts

```diff
@@ -0,0 +1,145 @@
+  it('should deny access, when no access key is provided', async () => {
+    const result = await fetch(basePathProtected + '/sample/aggregated');
+
+    expect(result.status).equals(403);
+    expect(result.headers.get('Content-Type')).equals('application/json;charset=ISO-8859-1');
+    expect((await result.json()).error).to.deep.equal({
+      detail: 'An access key is required to access /sample/aggregated.',
+      status: 403,
+      title: 'Forbidden',
+      type: 'about:blank',
+    });
+  });
+
+  it('should deny access, when wrong access key is provided', async () => {
+    const invalidAccessKey = 'invalidKey';
+    const result = await fetch(`${basePathProtected}/sample/aggregated?accessKey=${invalidAccessKey}`);
+
+    expect(result.status).equals(403);
+    expect(result.headers.get('Content-Type')).equals('application/json;charset=ISO-8859-1');
+    expect((await result.json()).error).to.deep.equal({
+      detail: 'You are not authorized to access /sample/aggregated.',
+      status: 403,
+      title: 'Forbidden',
+      type: 'about:blank',
+    });
+  });
+
+  it('should deny access, when providing aggregated access key on non aggregated route', async () => {
+    const result = await fetch(`${basePathProtected}/sample/details?accessKey=${aggregatedAccessKey}`);
+
+    expect(result.status).equals(403);
+    expect(result.headers.get('Content-Type')).equals('application/json;charset=ISO-8859-1');
+    expect((await result.json()).error).to.deep.equal({
+      detail: 'You are not authorized to access /sample/details.',
+      status: 403,
+      title: 'Forbidden',
+      type: 'about:blank',
+    });
+  });
+
+  it('should grant access, when providing aggregated access key', async () => {
+    const result = await fetch(`${basePathProtected}/sample/aggregated?accessKey=${aggregatedAccessKey}`);
+
+    expect(result.status).equals(200);
+  });
+
+  it('should grant access, when providing full access key', async () => {
+    const result = await fetch(`${basePathProtected}/sample/aggregated?accessKey=${fullAccessKey}`);
+
+    expect(result.status).equals(200);
+  });
+
+  it('should grant access, when providing full access key on non aggregated route', async () => {
+    const result = await fetch(`${basePathProtected}/sample/details?accessKey=${fullAccessKey}`);
+
+    expect(result.status).equals(200);
+  });
+});
+
+describe('Protected mode on POST requests', () => {
+  it('should deny access, when no access key is provided', async () => {
+    const result = lapisClientProtected.postAggregated1({ aggregatedPostRequest: {} });
+
+    result.catch(error => {
+      expect(error.status).equals(403);
+      expect(error.headers.get('Content-Type')).equals('application/json;charset=ISO-8859-1');
+      expect(error.error).to.deep.equal({
+        detail: 'An access key is required to access /sample/aggregated.',
+        status: 403,
+        title: 'Forbidden',
+        type: 'about:blank',
+      });
+    });
```

Comment:
This needs to make sure that the test doesn't pass if the response is 200 (providing a valid access key still yields a green test)